src/init/features/frameworkstack/index.ts

```diff
@@ -0,0 +1,69 @@
+import * as clc from "colorette";
+import * as utils from "../../../utils";
+import { logger } from "../../../logger";
+import { promptOnce } from "../../../prompt";
+import {
+  DEFAULT_REGION,
+  ALLOWED_REGIONS,
+  DEFAULT_DEPLOY_METHOD,
+  ALLOWED_DEPLOY_METHODS,
+} from "./constants";
+import { Options } from "../../../options";
+import { ensure } from "../../../ensureApiEnabled";
+import { Config } from "../../../config";
+import { requirePermissions } from "../../../requirePermissions";
+
+/**
+ * Setup new Frameworkstack project.
+ */
+export async function doSetup(setup: any, config: Config, options: Options): Promise<void> {
+  const projectId = setup?.rcfile?.projects?.default;
+  if (projectId) {
+    await requirePermissions({ ...options, project: projectId });
+    await Promise.all([
+      ensure(projectId, "firestack.googleapis.com", "unused", true),
+    ]);
+  }
+
+  setup.frameworkstack = {};
+
+  utils.logBullet("First we need a few details to create your service.");
+
+  await promptOnce(
+    {
+      name: "serviceName",
+      type: "input",
+      default: "acme-inc-web",
+      message: "Create a name for your service [1-64 characters]",
```

Comment:
It looks like we aren't validating this requirement at all. Can we validate it and reprompt if it doesn't meet our requirement?

Reply:
The reason to leave this is, this validation should happen on backend, similar to 
the case where a new `project id` is taken as input (it should be between 6 to 32 characters) and the error is shown when the `Creating Google Cloud Platform project` step is executed. WDYT?